src/main/java/me/zeroeightsix/kami/module/modules/movement/Guimove.java

```diff
@@ -0,0 +1,55 @@
+package me.zeroeightsix.kami.module.modules.movement;
+
+import me.zero.alpine.listener.EventHandler;
+import me.zero.alpine.listener.Listener;
+import me.zeroeightsix.kami.module.Module;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.client.gui.GuiChat;
+import net.minecraftforge.client.event.InputUpdateEvent;
+import org.lwjgl.input.Keyboard;
+
+@Module.Info(name = "Guimove", category = Module.Category.MOVEMENT, description = "Move while GUI is open")
+public class Guimove extends Module {
+    @EventHandler
+    private Listener<InputUpdateEvent> inputUpdateEventListener = new Listener<>(event -> {
+        if(mc.currentScreen != null && !(mc.currentScreen instanceof GuiChat) && mc.world != null && mc.player != null) {
+            if (Keyboard.isKeyDown(mc.gameSettings.keyBindForward.getKeyCode())) {
```

Comment:
Here is some pseudo-code for how you could probably clean this up.

```java
/*
 * MIT License
 * 
 * Copyright (c) 2020 0x22
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

private static final KeyBinding[] MOVE_KEYS = {
                    mc.gameSettings.keyBindForward,
                    mc.gameSettings.keyBindBack,
                    mc.gameSettings.keyBindLeft,
                    mc.gameSettings.keyBindRight,
                    mc.gameSettings.keyBindJump,
                    mc.gameSettings.keyBindSprint
            };

void onPreTick() {
    for (final KeyBinding bind : MOVE_KEYS) {
        final int keyCode = bind.getKeyCode();
        KeyBinding.setKeyBindState(keyCode , Keyboard.isKeyDown(keyCode));
    }
}
```

Reply:
What do you mean?? It's psudocode, of course you can't copy and paste it and have it work right away.

Reply:
I did a bit of testing and isKeyDown doesn't work in an inventory @0-x-2-2

Reply:
I was replying to 0x22

Reply:
> > I did a bit of testing and isKeyDown doesn't work in an inventory @0-x-2-2
> 
> ?

You are using this same method in your inventory move right now.

Reply:
It works in the mixin. It does not work on it's own.

Reply:
> It works in the mixin. It does not work on it's own.

There is no difference and the keys state will be the same as pre-tick because your hook is just called later in the tick and key polling only happens at the end of every render tick when the window updates and gets repainted.

Reply:
You completely lost me there. So what you're trying to tell me is that it doesn't work?

Reply:
no, I am saying the opposite of that.

Simple words: It works and this code is literally 1:1 from Future for this logic.

Reply:
See my comment in #681